cubi_tk/sodar/pull_data_collection.py

```diff
@@ -0,0 +1,364 @@
+        )
+
+        group_files = parser.add_mutually_exclusive_group(required=True)
+
+        group_files.add_argument(
+            "-p", "--preset", help="Preset to use for file selection.", choices=cls.presets.keys()
+        )
+        group_files.add_argument(
+            "-f", "--file-pattern", help="File pattern to use for file selection.", nargs="+"
+        )
+        group_files.add_argument(
+            "-a",
+            "--all-files",
+            action="store_true",
+            help="Do not filter files, download everything.",
+        )
+
+        group_samples = parser.add_argument_group("Sample Filters")
+        group_samples.add_argument(
+            "--substring-match",
+            action="store_true",
+            help="Defined samples do not need to match collections exactly, a substring match is enough",
+        )
+
+        group_samples.add_argument(
+            "-s",
+            "--sample-list",
+            nargs="+",
+            help="Sample list used for filtering collections."
+            "Takes precedence over --tsv and --biomedsheet.",
+        )
+        group_samples.add_argument(
+            "--biomedsheet",
+            help="Biomedsheet file for filtering collections. Sets tsv-column to 2 and "
+            "tsv-skip-rows to 12. Takes precedence over --tsv.",
+        )
+        group_samples.add_argument(
+            "--tsv", help="Tabular file with sample names to use for filtering collections."
+        )
+        group_samples.add_argument(
+            "--tsv-column",
+            default=1,
+            help="Column index for sample entries in tsv file. Default: 1.",
+        )
+        group_samples.add_argument(
+            "--tsv-skip-rows", default=0, help="Number of header lines in tsv file. Default: 0."
+        )
+
+        parser.add_argument(
+            "-o", "--output-dir", help="Output directory. Default: $PWD", default=os.getcwd()
+        )
+        parser.add_argument(
+            "--output-pattern",
+            default="{collection}/{subcollections}/{filename}",
+            help="Pattern for output files. Default: '{collection}/{subcollections}/{filename}'",
+        )
+        parser.add_argument(
+            "--output-regex",
+            nargs=3,
+            action="append",
+            metavar=("FILEPART", "MATCH", "REPL"),
+            default=[],
+            type=str,
+            help="Regular expression to change parts from iRODS path for output pattern. "
+            "Syntax: 'collection|subcollections|filename' 'regex' 'replacement'"
+            "Can be given multiple times, Default: None",
+        )
+
+        parser.add_argument(
+            "project_uuid",
+            help="SODAR project UUID",
+        )
+
+    @classmethod
+    def run(
+        cls, args, _parser: argparse.ArgumentParser, _subparser: argparse.ArgumentParser
+    ) -> typing.Optional[int]:
```

Comment:
In python 3.10+, `type | None` is preferred over (and equivalent to) `Optional[type]` -- as far as I know. But that's cosmetic anyways.
```suggestion
    ) -> int | None:
```